Src/PerformanceCollector/Shared/Implementation/QuickPulse/IQuickPulseServiceClient.cs

```diff
@@ -16,15 +17,33 @@ internal interface IQuickPulseServiceClient
         /// </summary>
         /// <param name="instrumentationKey">InstrumentationKey for which to submit data samples.</param>
         /// <param name="timestamp">Timestamp to pass to the server.</param>
+        /// <param name="configurationETag">Current configuration ETag that the client has.</param>
+        /// <param name="authApiKey">Authentication API key.</param>
+        /// <param name="configurationInfo">When available, the deserialized response data received from the server.</param>
         /// <returns><b>true</b> if data is expected, otherwise <b>false</b>.</returns>
-        bool? Ping(string instrumentationKey, DateTimeOffset timestamp);
+        bool? Ping(
+            string instrumentationKey,
+            DateTimeOffset timestamp,
+            string configurationETag,
+            string authApiKey,
+            out CollectionConfigurationInfo configurationInfo);
 
         /// <summary>
         /// Submits a data samples to QPS.
         /// </summary>
         /// <param name="samples">Data samples.</param>
         /// <param name="instrumentationKey">InstrumentationKey for which to submit data samples.</param>
+        /// <param name="configurationETag">Current configuration ETag that the client has.</param>
+        /// <param name="authApiKey">Authentication API key.</param>
+        /// <param name="configurationInfo">When available, the deserialized response data received from the server.</param>
+        /// <param name="collectionConfigurationErrors">Errors to be reported back to the server.</param>
         /// <returns><b>true</b> if the client is expected to keep sending data samples, <b>false</b> otherwise.</returns>
-        bool? SubmitSamples(IEnumerable<QuickPulseDataSample> samples, string instrumentationKey);
+        bool? SubmitSamples(
+            IEnumerable<QuickPulseDataSample> samples,
+            string instrumentationKey,
+            string configurationETag,
+            string authApiKey,
+            out CollectionConfigurationInfo configurationInfo,
+            CollectionConfigurationError[] collectionConfigurationErrors);
```

Comment:
These are different. 
Each sample represents a second-long data aggregate. The reason why there are more than one is to account for the fact that we might have failed to send a sample from the second before - and we want to backfill. In most cases, there will only be 1 sample sent out each second.
collectionConfigurationErrors represent all problems that were encountered by the SDK while parsing the current configuration. If, for instance, one of the metrics had a faulty filter that couldn't compile, we still want to collect the metric in accordance with other filters, and we still want to collect all other metrics, performance counters, and full telemetry documents, but we need to let the UX know that one of the filters failed. So each time we might be reporting any number of errors related to the currently active configuration.
Which is why there's no correlation between data and errors here.